texar/torch/modules/encoders/bert_encoder.py

```diff
@@ -328,7 +331,9 @@ def forward(self,  # type: ignore
                                      dtype=torch.int64)
         pos_embeds = self.position_embedder(sequence_length=pos_length)
 
-        if self.segment_embedder is not None:
+        if self.segment_embedder is not None and \
+                self.pretrained_model_name is not None and \
+                not self.pretrained_model_name.startswith('spanbert'):
```

Comment:
Why do we perform this check here? Wouldn't it make more sense to not construct a `segment_embedder` for SpanBERT?

Reply:
I agree.